### DISCUSSION TITLE
Is the full flow of theming and ResourceDictionary's well documented?

### DISCUSSION THREAD

**Discussion by JohnTasler** [unoplatform/uno#12662] [Q&A]
To understand the Visual Tree of XAML, it's important to see the default resources in the theme's generic.xaml, including `Color`, `Brush`, `Style`, and `ControlTemplate`. For UWP, I can find `generic.xaml` under `"C:\Program Files (x86)\"`, but I don't know where to look with Uno, especially with the `Uno.Toolkit.UI` controls, but also the WinUI 3 ones, too.

I've just looked through the [unoplatform/Uno.Themes on GitHub](https://github.com/unoplatform/Uno.Themes), but I still have a lot of confusion:

1. That project only appears to deal with Material and Cupertino themes. Where are the default resources for Fluent?
2. **It's very confusing** how you have so many **resource _classes_, rather than just XAML files**. It would be nice if your `ResourceDictionary` scheme was well-documented. I remember working with Xceed and other XAML controls back in the early days of WPF. It seems that every vendor has its own way of dealing with theming.

## Update
I think I was getting confused about how the Uno platform is packaged, vs. how Microsoft packages the UWP SDK. So, of course I can't find the `Style`s, etc. in a locally-installed folder or assembly. Instead I cloned a couple of Uno Platform repos, incuding `uno.extensions`, `Uno.Themes`, `uno.toolkit.ui`. This gets me much further along. If the DLL's only shipped with PDB's, then I could easily debug into the Un Platform code.

## However...
My point # 2 above is still valid and **probaby needs more informaton** about the theming and resources. It's complicated enough for 3rd party resource/theming for a _single_ platform, let alone the broad set that Uno supports. I say **probably** since I'm sure I haven't found and read all the Uno documentation yet.

I still haven't yet found the Fluent themes yet.

**Comment by jeromelaban** [comment]
Thanks for the questions!

We indeed have multiple locations for styles, fluent being here ([v1](https://github.com/unoplatform/uno/tree/master/src/Uno.UI.FluentTheme.v1) and [v2](https://github.com/unoplatform/uno/tree/master/src/Uno.UI.FluentTheme.v2)) and are almost one to one with the official styles from WinUI (certain details need to be adjusted in Uno for missing features). Thoses files are merged using [Uno.XamlMerge](https://github.com/unoplatform/uno.xamlmerge.task), which is a derivative of WinUI's generic.xaml merge tooling for performance considerations.

We're not distributing an equivalent of generic.xaml for those styles, and we're generally pointing users to the folders in the source tree for finding the original styles to override. Yet, it would make sense for us to document more formally how this process should be done.

For the **Uno.Themes**, the idea behind resource classes are related to being able to perform resource overrides in order to match material's guidelines ([Primary, secondary, etc..](https://github.com/unoplatform/Uno.Themes/blob/master/src/library/Uno.Material/Styles/Application/v2/SharedColors.xaml)) as well a font overrides. We need to have classes because of [conditional overrides](https://platform.uno/docs/articles/external/uno.themes/doc/material-getting-started.html#manual-color-overrides) provided by users (and could be generated by the [DSP tooling](https://platform.uno/docs/articles/external/uno.themes/doc/material-getting-started.html#color-overrides-using-material-theme-builder-and-dsp-format) using [Material Theme Builder](https://m3.material.io/theme-builder#/custom)), and code makes it easier to handle restrictions on WinUI's handling of resource dictionaries. Note that Uno.Themes is about re-theming out-of-the-box controls from WinUI ([styles are here](https://github.com/unoplatform/Uno.Themes/tree/master/src/library/Uno.Material/Styles/Controls/v2)).

**Uno.Toolkit.UI** is different, where the library is about adding new controls, like [Cards](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/controls/CardAndCardContentControl.html) or [Chips](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/controls/ChipAndChipGroup.html), which also have their own theming for fluent and material where appropriate. This library only defines styles for its own controls.

**Reply by kazo0** [reply]
Good points! As Jerome mentioned, the Uno.Theme library is meant to provide a collection of styles for Material/Cupertino that can be used _instead_ of the default Fluent styles coming from Uno/WinUI. 

I can understand the confusion surrounding the many resource classes that exist right now in Themes. If we take Uno.Material as an example, there are many resource classes at the moment that are mostly present in order to allow for backwards compatibility with the  v1 (Material Design 2) version of the library. As of today, the only resource class that is relevant is the [MaterialTheme](https://github.com/unoplatform/Uno.Themes/blob/master/src/library/Uno.Material/MaterialTheme.cs) class, which performs all the coordination of the required resource dictionaries during initialization

As for docs, I'd point you towards the [Themes overview](https://platform.uno/docs/articles/external/uno.themes/doc/themes-overview.html) if you haven't found that already and if you wanted more info about the MaterialTheme resource class, that would be found [here](https://platform.uno/docs/articles/external/uno.themes/doc/material-getting-started.html#materialtheme).

Let me know if you need more info!
